### ISSUE TITLE
Dialog not closing when connection opened

### ISSUE THREAD

**Issue by CalebWeaver** [angular/material2#5482]
#### Bug, feature request, or proposal:
When calling dialogRef.close() on an open dialog window, the background overlay disappears but the dialog stays opened. This only happens when a jsforce connection is opened.

#### What is the expected behavior?
The dialog window and background should both close when dialogRef.close() is called.

#### What is the current behavior?
Only the background overlay is removed. The dialog window stays opened.

#### What are the steps to reproduce?
The only way I've found to reproduce this is to open a connection to Salesforce using jsforce as a resolver on a route. Once this is done, opening a popup on the route and attempting to close it using dialogRef.close() should result in the dialog remaining open.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Material 2 beta 7, Angular 4.2.3

#### Is there anything else we should know?
This is referencing the current version of jsforce (https://github.com/jsforce/jsforce)

**Comment by jelbourn** [comment]
Sounds like something is escaping the zone. We probably won't be able to reproduce this ourselves. If you inject `NgZone` and wrap your call to close the dialog in `ngZone.run(() => {  /** close here */ })`, does it resolve the issue?

**Comment by CalebWeaver** [comment]
Yes!
`this.zone.run(() => this.dialogRef.close());`
Closes the dialog successfully.

**Comment by jelbourn** [comment]
@CalebWeaver sounds like a bug in [zone.js](https://github.com/angular/zone.js/), then. Zone is supposed to know when any kind of async event occurs, which is used by Angular to run change detection. Something in jsforce must be using an API zone isn't aware of / capturing. Closing this issue since it's out of scope for Angular Material, but feel free to open an issue on the zone repo.
